libfreerdp/core/listener.c

```diff
@@ -223,6 +223,27 @@ static BOOL freerdp_listener_open_local(freerdp_listener* instance, const char*
 #endif
 }
 
+static BOOL freerdp_listener_open_from_socket(freerdp_listener* instance, int fd)
+{
+#ifndef _WIN32
+	rdpListener* listener = (rdpListener*) instance->listener;
+
+	if (fcntl(fd, F_SETFL, O_NONBLOCK) < 0)
+		return FALSE;
+
+	listener->sockfds[listener->num_sockfds] = fd;
+	listener->events[listener->num_sockfds] = CreateFileDescriptorEvent(NULL, FALSE, FALSE, fd);
+	if (!listener->events[listener->num_sockfds])
+		return FALSE;
+
+	listener->num_sockfds++;
+
```

Comment:
It might be important to check listener->num_sockfds against MAX_LISTENER_HANDLES in the beginning of the function - right ?